pkg/cmdutil/auth_check.go

```diff
@@ -18,7 +18,7 @@ func CheckAuth(cfg config.Config) bool {
 	// authentication tokens set for enterprise hosts.
 	// Any non-github.com hostname is fine here
 	dummyHostname := "example.com"
-	token, _ := cfg.Authentication().Token(dummyHostname)
+	token, _ := cfg.Authentication().TokenFromEnvOrConfig(dummyHostname)
```

Comment:
Since we don't need an actual token, could we come up with an approach to just test whether an environment variable was set? We used to go the token route in the past because it was cheap, but now with keyring it's not so cheap anymore, so instead of coming up with workarounds I would prefer to just see an approach that specifically asks for the information we're interested in (and nothing more).

I'm imagining a method called something like `IsEnterpriseTokenEnv() bool`

Reply:
😅 that was my initial approach but changed following @samcoe's comment. He pointed out this could be used also [here](https://github.com/cli/cli/pull/7169/files#diff-365859dee5ab3df41ac4cfa3c2ccc39bba2cfbbc288a56387c439f7b8558eab8L89). Both are fine with me, let me know what is prefered

Reply:
Both call points are only interested in whether GH_ENTERPRISE_TOKEN was set, but not interested in the value itself. I'd vote for a new method that is precisely tuned to what is being asked for.

Reply:
maybe something to export from https://github.com/cli/go-gh?

Reply:
I am okay with a method `HasEnterpriseEnvToken() bool`. Sorry for the back and forth on this. I don't think we need to expose any additional functionality in `go-gh` to achieve this. The `ghAuth.TokenFromEnvOrConfig` function returns the source of the token as its second return value. We just need to verify that the source is either `GITHUB_ENTERPRISE_TOKEN` or `GH_ENTERPRISE_TOKEN`.

Reply:
@mislav I didn't notice at first, but this is not actually specific to Enterprise. If there is an Enterprise token, it will be returned, true. But if there is no enterprise token, it will fallback to `GH_TOKEN`, `GITHUB_TOKEN`.

My proposal is to have the original behaviour (`TokenFromEnvOrConfig() != ""`), wrapped in a function `HasEnvOrConfigToken` to encapsulate this logic

Reply:
see https://github.com/cli/cli/pull/7169/commits/70a6acc2fd2501abfce69c89299ba8fb8c7665c4